core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -584,6 +584,38 @@ class DAGScheduler(
       callSite: CallSite,
       resultHandler: (Int, U) => Unit,
       properties: Properties): JobWaiter[U] = {
+     submitJob[T, U](rdd, (_: Int) => {}, func, partitions, callSite, resultHandler, properties)
+  }
+
+  /**
+   * Create configs on driver as part of commit protocol using a provided function that accepts
+   * an integer (stageId) as parameter and then Submit an action job to the scheduler
+   *
+   * @param rdd target RDD to run tasks on
+   * @param jobSetupFunc This function gets called on the driver after the final stage is known.
+   *   Its primary purpose is as part of the commit protocol, since that needs
+   *   to know the stageId for creating a unique id. See SparkHadoopWriter
+   * @param func a function to run on each partition of the RDD
+   * @param partitions set of partitions to run on; some jobs may not want to compute on all
+   *   partitions of the target RDD, e.g. for operations like first()
+   * @param callSite where in the user program this job was called
+   * @param resultHandler callback to pass each result to
+   * @param properties scheduler properties to attach to this job, e.g. fair scheduler pool name
+   *
+   * @return a JobWaiter object that can be used to block until the job finishes executing
+   *         or can be used to cancel the job.
+   *
+   * @throws IllegalArgumentException when partitions ids are illegal
+   */
+  private[spark]
+  def submitJob[T, U](
```

Comment:
again, don't think you need to add another version of this ... I think you just need to change the signature of the existing one, and then change all callsites.

Reply:
I actually did that. In the original submitJob, I am calling the new private one